client/api/src/lib.rs

```diff
@@ -85,6 +85,7 @@ pub mod utils {
 				}
 			}
 
+			// TODO: USE THIS!!!
 			let ancestor = sp_blockchain::lowest_common_ancestor(client, *hash, *base)?;
```

Comment:
Instead of relying on [BlockHash<T: Config> = StorageMap<_, Twox64Concat, T::BlockNumber, T::Hash, ValueQuery>](https://github.com/paritytech/substrate/blob/50176e1b57bdf6168d9bf3764d1cce9ed8bcf611/frame/system/src/lib.rs#L566) mapping, we could use `sp_blockchain::lowest_common_ancestor()` from client context to find the ancestor block hash.